data_safe_haven/config/sre_config.py

```diff
@@ -18,8 +17,8 @@
 
 
 def sre_config_name(sre_name: str) -> str:
-    """Construct a safe YAML filename given an input SRE name."""
-    return f"sre-{json_safe(sre_name)}.yaml"
+    """Construct a YAML filename given an input SRE name."""
+    return f"sre-{sre_name.lower()}.yaml"
```

Comment:
Are we worried this will introduce the same kind of problem?

We are handling `sre_name` differently across the code?

Reply:
_i.e._, should the validator insist lowercase, or should we allow uppercase in the file names?

Reply:
+1 - let the validator do this work.

Reply:
Added a new validator, as SafeString is used for docker PATs (among other things) and thus needs to accept upper case